python/graphstorm/trainer/glem_np_trainer.py

```diff
@@ -158,17 +158,25 @@ def fit(self, train_loader, num_epochs,
             t0 = time.time()
             rt_profiler.start_record()
 
-            if freeze_input_layer_epochs <= epoch:
-                self._model.lm.unfreeze_input_encoder()
-                no_pl = False
-
             use_gnn = self._model.em_order_gnn_first
             # `use_gnn`` determines which part to train, if `em_order_gnn_first`
             # 1st round: train GNN, fix LM; 2nd round: train LM fix gnn
             for _ in range(2):
                 stage_start_time = time.time()
                 part_to_train = 'gnn' if use_gnn else 'lm'
                 self._model.toggle(part_to_train)
+                if part_to_train == 'gnn' and not self._model.node_input_encoder.use_cache:
```

Comment:
the code here is difficult to understand. I think we need to handle the cache correctly in `freeze_input_encoder`, instead of handling here.

when freezing the input encoder, we need to check the existence of the cache and update the cache if the LM model is updated. However, i think we should clear up the cache in `toggle` when we switch from GNN to LM.

Reply:
Moved to `update_cache`